docs/pipelineruns.md

```diff
@@ -844,6 +847,378 @@ For more information, see the following topics:
+  workspaces:
+  - name: shared-data
+    persistentVolumeClaim:
+      claimName: shared-task-storage
+```
+
+Upon execution, this will cause failures:
+
+```yaml
+# Failed execution of the above PipelineRun
+
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: recipe-time-
+  ...
+spec:
+  pipelineRef:
+    name: fetch-and-print-recipe
+  workspaces:
+  - name: shared-data
+    persistentVolumeClaim:
+      claimName: shared-task-storage
+status:
+  completionTime: "2022-06-02T19:02:58Z"
+  conditions:
+  - lastTransitionTime: "2022-06-02T19:02:58Z"
+    message: 'Tasks Completed: 1 (Failed: 1, Canceled 0), Skipped: 1'
+    reason: Failed
+    status: "False"
+    type: Succeeded
+  pipelineSpec:
+    ...
+  taskRuns:
+    recipe-time-v5scg-fetch-the-recipe:
+      pipelineTaskName: fetch-the-recipe
+      status:
+        completionTime: "2022-06-02T19:02:58Z"
+        conditions:
+        - lastTransitionTime: "2022-06-02T19:02:58Z"
+          message: |
+            "step-fetch-and-write" exited with code 1 (image: "docker.io/library/ubuntu@sha256:26c68657ccce2cb0a31b330cb0be2b5e108d467f641c62e13ab40cbec258c68d"); for logs run: kubectl -n default logs recipe-time-v5scg-fetch-the-recipe-pod -c step-fetch-and-write
+          reason: Failed
+          status: "False"
+          type: Succeeded
+        ...
+        taskSpec:
+          steps:
+          - image: ubuntu
+            name: fetch-and-write
+            resources: {}
+            script: | # See below: Replacements do not happen.
+      	      echo hi >> $(workspaces.shared-data.path)/recipe.txt
+```
+
+##### Embedded Specifications with Referenced Resources
+
+In the case of embedded specifications that have references, the `Workspaces` will be propagated up to the embedded specification level only. Users need to explicitly declare `Workspaces` wherever they reference resources.
+
+```yaml
+# PipelineRun attempting to propagate Workspaces to referenced Tasks
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: fetch-secure-data
+spec:
+  workspaces: # If Referenced, Workspaces need to be explicitly declared
+  - name: recipe-store
+  - name: shared-data
+  - name: password-vault
+  steps:
+  - name: fetch-and-write
+    image: ubuntu
+    script: |
+      if [ "hunter2" = "$(cat $(workspaces.password-vault.path)/password)" ]; then
+        cp $(workspaces.recipe-store.path)/recipe.txt $(workspaces.shared-data.path)
+      else
+        echo "wrong password!"
+        exit 1
+      fi
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: recipe-time-
+spec:
+  workspaces:
+  - name: password-vault
+    secret:
+      secretName: secret-password
+  - name: recipe-store
+    configMap:
+      name: sensitive-recipe-storage
+      items:
+      - key: brownies
+        path: recipe.txt
+  - name: shared-data
+    persistentVolumeClaim:
+      claimName: shared-task-storage
+  pipelineSpec:
```

Comment:
Can this example be simplified at all? Maybe just one task in the Pipeline, which is referenced, and only one workspace.

Reply:
Yes. I can do that.